mspy_vendi/domain/impressions/filters.py

```diff
@@ -0,0 +1,26 @@
+import datetime
+
+from pydantic import NonNegativeInt
+
+from mspy_vendi.core.filter import BaseFilter
+from mspy_vendi.domain.impressions.models import Impression
+
+
+class ImpressionFilter(BaseFilter):
+    id__in: list[NonNegativeInt] | None = None
```

Comment:
```suggestion
    id__in: list[PositiveInt] | None = None
    
```
as id=0 is invalid input